agent/structs/structs.go

```diff
@@ -1045,6 +1059,8 @@ func (c *HealthCheck) IsSame(other *HealthCheck) bool {
 func (c *HealthCheck) Clone() *HealthCheck {
 	clone := new(HealthCheck)
 	*clone = *c
+	clone.ServiceTags = cloneStringSlice(c.ServiceTags)
+	clone.Definition = *c.Definition.Clone()
```

Comment:
I wonder if this get's called in any super hot paths that could negatively affect performance? For example do we ever call this during service discovery (/health) calls?

Reply:
I removed the changes in `agent/structs` and everything passed the race detector so these fields being deep copied was overkill on my part.